src/Admin/OptionsMenu.php

```diff
@@ -165,6 +166,63 @@ public function screen_handle() {
 		return sprintf( 'toplevel_page_%s', AMP_Options_Manager::OPTION_NAME );
 	}
 
+	/**
+	 * Registers shimmed assets not guaranteed to be available in core.
+	 */
+	public function register_shimmed_assets() {
+		if ( ! wp_script_is( 'wp-api-fetch', 'registered' ) ) {
+			$asset_handle = 'wp-api-fetch';
+			$asset_file   = AMP__DIR__ . '/assets/js/' . $asset_handle . '.asset.php';
+			$asset        = require $asset_file;
+			$version      = $asset['version'];
+
+			wp_register_script(
+				$asset_handle,
+				amp_get_asset_url( 'js/' . $asset_handle . '.js' ),
+				[],
+				$version,
+				true
+			);
+
+			wp_add_inline_script(
+				$asset_handle,
+				sprintf(
+					'wp.apiFetch.use( wp.apiFetch.createRootURLMiddleware( "%s" ) );',
+					esc_url_raw( get_rest_url() )
+				),
+				'after'
+			);
+			wp_add_inline_script(
+				$asset_handle,
+				implode(
+					"\n",
+					array(
+						sprintf(
+							'wp.apiFetch.nonceMiddleware = wp.apiFetch.createNonceMiddleware( "%s" );',
+							( wp_installing() && ! is_multisite() ) ? '' : wp_create_nonce( 'wp_rest' )
+						),
+						'wp.apiFetch.use( wp.apiFetch.nonceMiddleware );',
+						'wp.apiFetch.use( wp.apiFetch.mediaUploadMiddleware );',
+						sprintf(
+							'wp.apiFetch.nonceEndpoint = "%s";',
+							admin_url( 'admin-ajax.php?action=rest-nonce' )
+						),
+					)
+				),
+				'after'
+			);
+		}
+
+		if ( ! wp_style_is( 'wp-components', 'registered' ) ) {
```

Comment:
If churn in the core `wp-components` CSS causes issues with our styles down the line, we may want to use our copy of `wp-components` on AMP plugin screens in all WP versions.